### PR TITLE
Keith 2024 01 18 d

### PR DESCRIPTION
**Pull request by keithgibson**
The comments from my commits. Not sure if those are brought to the fore, so reposting (with some edits) here as there's a lot of info:

Refactored state for no more need for separate stores for the all-time vs byYear for each category (artist, track, album). Instead, one store for each, and everything goes into the object at the key of the year, save for all-time which is at all-time. The actions for each look for the year passed in. So we always pass a year argument to each fetch action, e.g. fetchTopTracks(year), and no longer need to do 'query:year' or anything like that. If the year is anything other than 2024 (which it is when the slider is at the far right), then the ByYear router is activated as usual: the logic from the router onwards to the models remains the same for each category. And now all the Components (Top Tracks, Top Artists) can be categorized generally as Top rather than 'Top' and 'byYear' both, because they all show all-time data and byYear data at different times. So the 'Top' is referring to the 'top' of the time period chosen, be it one year or all-time. Year is set state when the sliderComp is clicked. We also dispatch the update of each store in sliderComp: TopTracks, TopArtists, and (soon) Top Album(s). The most recent data is placed in the arrData field of the respective store, then each component subscribes to that field in its store to get the most recent data that should be displayed.

Also, the tracks list no longer re-renders and the track list and track image elements don't move when a year is clicked. Not sure if it's because of something I did or if it was already fixed when I pulled down dev most recently. Still getting the bug of all of the tracks marqueeing when played, but only after a year is clicked. ¯\_(ツ)_/¯

### CODE REVIEW

File: client/src/components/TopArtistsComp.jsx
Comment (rterker) [inline client/src/components/TopArtistsComp.jsx:1]: <></> is shorthand for React.fragment
key should be unique for every element in a list. artist.name isn't unique
also, this is an old version of this file. i haven't merged in the new one

File: client/src/components/SliderComp.jsx
Comment (rterker) [inline client/src/components/SliderComp.jsx:1]: maybe it's my style, but i'm not sure why fetching dispatches in a wrapper function is necessary here. fetchData is only being used right above it (not being called in other files), so all it really does is make you look back and forth for what is actually happening here. it feels unnecessary. NBD, but food for thought.

**Review by rterker — COMMENTED** [review]
I get the benefit of improving names, but it feels dangerous to be changing so many names and endpoints at this point in the game, unless really necessary, especially when other components have them as dependencies. It opens up the possibility for bugs in a lot of places. Can we stick with the current names moving forward?

File: client/src/components/SliderComp.jsx
Comment (keithgibson) [reply client/src/components/SliderComp.jsx:1]: it's used twice. on idle, i.e. when the page loads, and when the year is clicked.

File: client/src/components/TopArtistsComp.jsx
Comment (keithgibson) [reply client/src/components/TopArtistsComp.jsx:1]: the shorthand doesn’t support keys. i was getting warnings in console that each mapped element should have a key. when i used react.fragment, that warning was resolved.

error:
"
react.development.js:209 Warning: Each child in a list should have a unique "key" prop.
Check the render method of `TopArtistsComp`.
"

**Comment by keithgibson** [comment]
closing because i cant push changes to this branch due to the open pr.

**Pull request closed without merging**